core.spec.md

```diff
@@ -0,0 +1,501 @@
+    - *then* **Return** `coreName` = {d}'s name
+- If no matching directive was found, the ***Has Core Feature* validation fails**.
+
+## Feature Collection (validates Name Uniqueness)
+
+Collect a map of ({featureName}: `String`) -> `Directive`, where `Directive` is a {@core} Directive which introduces the feature named {featureName} into the document.
+
+CollectFeatures(document) :
+  - Let {coreName} be the name of the core feature found via {Bootstrap(document)}
+  - Let {features} be a map of {featureName}: `String` -> `Directive`, initially empty.
+  - For each directive {d} named `coreName` on the SchemaDefinition within {document},
+    - Let {name} be the spec's [name](#sec-Prefixing) as specified by the directive's [`as:`](#@core/as) argument or, if the argument is not present, the default name from the [feature url](#@core/feature).
+    - If {name} exists within {features}, the ***Name Uniqueness* validation fails**.
+    - Insert {name} => {d} into {features}
+  - **Return** {features}
+
+
+Prefixes, whether implicit or explicit, must be unique within a document. Valid:
+
+:::[example](prefixing.graphql#schema[0]) -- Unique prefixes
+
+It is also valid to reference multiple versions of the same spec under different prefixes:
+
+:::[example](prefix-uniqueness.graphql#schema[0]) -- Explicit prefixes allow multiple versions of the same spec to coexist within a Document
+
+Without the explicit [`as:`](#@core/as), the above would be invalid:
+
+:::[counter-example](prefix-uniqueness.graphql#schema[1]) -- Non-unique prefixes with multiple versions of the same spec
+
+Different specs with the same prefix are also invalid:
+
+:::[counter-example](prefix-uniqueness.graphql#schema[2]) -- Different specs with non-unique prefixes
+
+## Assign Features
+
+Create a map of {element}: *Any Named Element* -> {feature}: `Directive` | {null}, associating every named schema element within the document with a feature directive, or {null} if it is not associated with a feature.
+
+AssignFeatures(document) :
+  - Let {features} be the result of collecting features via {CollectFeatures(document)}
+  - Let {assignments} be a map of ({element}: *Any Named Element*) -> {feature}: `Directive` | {null}, initally empty
+  - For each named schema element {e} within the {document}
+    - Let {name} be the name of the {e}
+    - If {name} begins with {"__"},
+      - Insert {e} => {null} into {assignments}
+      - **Continue** to next {e}
+    - If {name} contains the substring {"__"},
+      - Partition {name} into `[`{prefix}, {base}`]` at the first {"__"} (that is, find the shortest {prefix} and longest {base} such that {name} = {prefix} + {"__"} + {base})
+      - If {prefix} exists within {features}, insert {e} => {features}`[`{prefix}`]` into {assignments}
+        - Else, insert {e} => {null} into {assignments}
+      - **Continue** to next {e}
+    - Insert `e => null` into {assignments}
```

Comment:
SUBSTANTIVE: This does not appear to handle root directives!

Reply:
Fixed